lib/CompatibilityPlugin.js

```diff
@@ -11,11 +11,13 @@ const NullFactory = require("./NullFactory");
 class CompatibilityPlugin {
 
 	apply(compiler) {
-		compiler.plugin("compilation", (compilation, params) => {
+		compiler.hooks.compilation.tap("CompatibilityPlugin", (compilation, {
+			normalModuleFactory
+		}) => {
 			compilation.dependencyFactories.set(ConstDependency, new NullFactory());
 			compilation.dependencyTemplates.set(ConstDependency, new ConstDependency.Template());
 
-			params.normalModuleFactory.plugin("parser javascript/auto", (parser, parserOptions) => {
+			normalModuleFactory.plugin("parser javascript/auto", (parser, parserOptions) => {
```

Comment:
For info: `normalModuleFactory.plugin("parser javascript/auto",` transforms into:

`normalModuleFactory.hooks.parser.tap("javascript/auto", "CompatibilityPlugin",`

But this can be done in a separate PR.

Reply:
I was going to ask about this also.